liquidity_pool/src/test.rs

```diff
@@ -32,7 +33,8 @@ fn generate_keypair() -> Keypair {
 }
 
 fn create_token_contract(e: &Env, id: &[u8; 32], admin: &Keypair) -> Token {
-    register_token(&e, id);
+    let contract_id = BytesN::from_array(e, &id);
+    e.register_contract_wasm(&contract_id, token_contract::WASM);
     let token = Token::new(e, id);
     // decimals, name, symbol don't matter in tests
     token.initialize(&to_ed25519(&e, admin), 7, "name", "symbol");
```

Comment:
I'd like to replace `token`, i.e. `soroban_token_contract::testutils::Token`, with the generated client that is in `token_contract::ContractClient`. Right now the former has a combination of helpers for calling a contract, as well as helpers for producing the right types of messages for those calls. I think we should split that in half, so that the only utilities a consuming contract needs to import as code is those utilities, not the calling functions.

We can then assess whether those utilities can be generalized into the SDKs at all.

Reply:
I'm pretty unclear on what you actually want to change here. Can you pick a specific function (eg `soroban_token_contract::testutils::Token::approve`) and explain what you want to be different in terms of implementation/usage/API?

Reply:
We discussed and going to revisit this. We can't do this now because there's too much convenience logic in the Token type that isn't included in the contract client, but @jonjove has some ideas about how we could generate additional clients with that functionality.